src/seedu/addressbook/data/person/Name.java

```diff
@@ -43,6 +45,42 @@ public static boolean isValidName(String test) {
         return Arrays.asList(fullName.split("\\s+"));
     }
 
+    /**
+     * Returns true if the other name is very similar to this name.
+     * Two names are considered similar if ...
+     */
+    public boolean isSimilar(Name other) {
+        if(other == null) {
+            return false;
+        }
+        List<String> wordsInOtherName = other.getWordsInName().stream()
+                .map(word -> word.toLowerCase())
+                .collect(Collectors.toList());
+        List<String> wordsInThisName = this.getWordsInName().stream()
+                .map(word -> word.toLowerCase())
+                .collect(Collectors.toList());
+
+        for (String wordOther : wordsInOtherName) {
+            for (String wordThis : wordsInThisName) {
+                if(wordThis.length() <= wordOther.length()) {
+                    if (wordOther.contains(wordThis)) {
+                        return true;
+                    }
+                } else { // wordOther is shorter
+                    if (wordThis.contains(wordOther)) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+
+        // put list to lower case first, takes care of case sensitivity
+        // take short one then long.contains(short), checks if shorter one is a substring
+        // take the words out and check if any words between short and long are similar
```

Comment:
Have not reached this LO yet but you can refer to W7.7 for how to write good comments. You should explain **what** and **why**, not how your code do it. Also you can refer to addressbook and think about where is the appropriate place to put comments. It might to be a good idea to put a chunk of comments at the end.

Reply:
If I may clarify, what does it mean to leave a chunk of comments at the end?

Reply:
I just think it would be preferable to put them before method declaration. You can checkout Javadoc standard comments and look through our addressbook codebase to see where they normally put comments